pkg/component/observability/monitoring/kubestatemetrics/resources.go

```diff
@@ -37,6 +37,156 @@ import (
 	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
 )
 
+/*
+TODO(vicwicker): The documentation instructions does not include the kind but the examples do. A quick test
+showed the kind is superfluous when --custom-resource-state-config is given. We need a bit more investigation
+in case there is a resource of kind CustomResourceStateMetrics that we could use to pass this lenghty
+configuration more elegantly.
+*/
```

Comment:
Have you had a look [at the API](https://pkg.go.dev/k8s.io/kube-state-metrics/v2@v2.12.0/pkg/customresourcestate)? Looks like it should be possible to avoid string literals and do it in code :)

Reply:
Yes. I moved to generate this configuration programatically some commits later :)

Reply:
Oops, that's what I get for going through the commits but stopping short of the end.